### DISCUSSION TITLE
Provide a backup system for those whose installation of SA stops functioning

### DISCUSSION THREAD

**Discussion by DNin01** [ScratchAddons/ScratchAddons#5291] [Discussion]
Based on https://github.com/ScratchAddons/ScratchAddons/issues/5289, SA can, on rare occasions, suddenly stop functioning, displaying no content in its popup or full screen settings. I'm not sure why, maybe an update could've done something weird, maybe cloud sync could've messed something up, who knows? But I do know that there isn't much you can do in such a situation - you can't even get to where you can export your settings file.

To solve this, we could come up with a feature or features to help users recover from SA crashes, like:
- Occasionally making an automatic backup of the settings file, and then offering a rollback if it becomes corrupted.
- Allowing more control (like entering SA's options so you can export/import your settings) or at least let the user know what's going on if something fails to load.
![A mockup. A warning message saying addons failed to load with a button below for recovery options.](https://user-images.githubusercontent.com/106490990/198857955-c9d3bb48-5818-4aa1-b51c-2847310e9e8f.png)

We can continue the discussion and think of any other ways we could tackle this problem, as well as troubleshoot it.

**Comment by DNin01** [comment]
By the way, feel free to move this discussion to a different category if you think it fits better somewhere else. I'm not sure if this fits best in Ideas or Discussion.

**Comment by DNin01** [comment]
Turns out, I can navigate to More Settings even if the addons list doesn't load, so that can be recovered from, although I don't think anyone would run into that issue in unmodified SA.

Only other explanation I have to #5289 is that maybe their settings file got corrupted in a cloud sync, update, or something like that. Keeping a locally stored, periodically updated backup of the settings file that we can let users access somehow should SA crash like that might be able to help get their settings back.

**Comment by GrahamSH-LLK** [comment]
We don't really have a place to store automated backups. Background page crashing is probably fixable with a browser restart. I doubt anything actually corrupted. Looking at the error in #5289 it _looks_ like a chromium bug given the lock error.
